### PR TITLE
Fix build

### LINKED ISSUE #38: Linux install with snap
link: https://dvc.org/doc/install/linux#install-with-snap

The additional flags `--beta` and `--edge` result in messages: 
- `Channel latest/beta for dvc is closed; temporarily forwarding to stable.`
- `Channel latest/edge for dvc is closed; temporarily forwarding to stable.`

- [x] Should mention of these flags remain if they are closed?

### PR DESCRIPTION
**Pull request by casperdcl**
TL;DR: fixes #38 and probably #3 & #4. Also makes #2 easy to fix in a follow-up too.

---

Holymoly, the [official docs](https://snapcraft.io/docs/python-apps) are worse than vague. (Particularly for [`plugin: python`](https://snapcraft.io/docs/python-plugin) && `confinement: classic`: "It is required to bundle python" yez but hau!?) Anyway having [searched hard](https://github.com/search?p=2&q=filename%3Asnapcraft.yaml+%22plugin%3A+python%22+%22base%3A+core20%22+%22confinement%3A+classic%22&type=Code) and seen multiple inconsistencies in canonical's own usage [[1](https://github.com/ubuntu/ubuntu-make/blob/81194cca29eb19865a0d49a70b6dc191027af64e/snap/snapcraft.yaml),[2](https://github.com/canonical/rockcraft/blob/be6dec45e26b4ed442fb493665c738d5f425131c/snap/snapcraft.yaml),[3](https://github.com/canonical/hotsos/blob/2ca2193669307dd599f34eb1bc0f62bb5c7ae497/snap/snapcraft.yaml),[4](https://github.com/canonical/charmcraft/blob/2575e866ab379c1c73c78960a86cd70d1aca7780/snap/snapcraft.yaml),[5](https://github.com/canonical/joule/blob/01ec440f5ff0cd1cda185c087afd6f02f360bcc8/snap/snapcraft.yaml),[6](https://github.com/canonical/magma-access-gateway-snap/blob/9d1003ae4c0d31d0433a3c49a82b762533314653/snap/snapcraft.yaml)] I ~need to lie down~ prevailed.

- [x] upgrade Ubuntu 18.04 -> 20.04 based images
- [x] accordingly upgrade Python to 3.8
- [x] bundle Python properly
- [x] fix `PATH` issues (use bundled Python & Git properly)
- [x] replace overkill `freeze.py` & associated linters with 3 lines of bash & yaml
- [x] drop Python & requirements from GH actions (snap should handle & bundle it all, ignoring the host system)
- [x] rebase & clean commit history (will be useful for tackling #2 later)

### CODE REVIEW

**Comment by casperdcl** [comment]
Merging because everything was broken before anyway so nothing to lose :)
